### DISCUSSION TITLE
Multiple stream vod folder in s3 when internet fluctuates, when internet fluctuates i need the last frame to be shown until stream restores.

### DISCUSSION THREAD

**Discussion by iamdinker1** [OvenMediaLabs/OvenMediaEngine#1316] [Help]
Hi currently, i am facing multiple internet fluctuations, so when I am saving vod for my stream in realtime i am facing this issue where multiple folders are being created when the stream stops and restores. I want it in such a way that the stream continues both live and VOD and stays as one stream when I stop the live channel. 

Also, one more thing i want to do is that when the internet fluctuates i want to show the last frame received and then resume as soon as feed is recieved. 
I would like to know if it exists or if this will be possible. Thank you.

**Comment by bchah** [comment]
For the first issue, I don't know how OME could determine the difference between a stream disconnecting due to internet fluctuation versus just unplugging an encoder or some other non-clean disconnect. I guess there could be a "persistent stream" type created by API where OME keeps the last frame repeating to the player all the time and the stream only stops via API. That would be a huge undertaking though, so let's zoom out a bit to the real problem - your internet fluctuations. Have you considered using a service like [Restream](restream.io) or [StreamYard](streamyard.com) to create a persistent live stream for you? That way you can keep a 100% consistent feed running to OME and still control the content even if your internet is going crazy.

As for your other request, this won't be something you need to chase anymore if you have your stream originating from a hosted 3rd-party. But for fun... the thumbnail publisher could help you here - you could cache the most recent thumbnail (say every 0.5s) and listen for a disconnection event from OvenPlayer using the events API, then overlay the thumbnail image on the player - clearing it when the player begins again 🧪🔬👨🏻‍🔬👨🏻‍🔬👨🏻‍🔬

**Comment by iamdinker1** [comment]
Won't using restream add extra delay in the stream? I want to achieve 2-4 second latency. + Restream will also add overhead costs since i will be trying to use it for 50-100 streams every day, maybe more.
